src/FunctionApps/BulkImport/Import/main.py

```diff
@@ -0,0 +1,98 @@
+import logging
+import os
+import zipfile
+
+import requests
+import multiprocessing
+import json
+from requests.adapters import HTTPAdapter
+from requests.packages.urllib3.util.retry import Retry
+
+unzipped_directory = "./FhirResources"
+
+def main(zip_file):
+    unzip_input_file(zip_file)
+    process_ndjson_files(unzipped_directory)
+
+def unzip_input_file(zip_file):
+    with zipfile.ZipFile(zip_file,"r") as zip_ref:
+        zip_ref.extractall(unzipped_directory)
+
+def process_ndjson_files(dir):
+    with multiprocessing.Pool(processes = multiprocessing.cpu_count()-1) as p:
+        directory = os.fsencode(dir)
+        for file in os.listdir(directory):
+            filename = os.fsdecode(file)
+            if filename.endswith(".ndjson"):
+                file_path = (os.path.join(dir, filename))
+                p.apply_async(read_file, (file_path,))
+            else:
+                logging.error(f"File must be ndjson, file given: {filename}")
+        p.close()
+        p.join()
+
+def read_file(file):
+    with open(file) as fp:
+        for line in fp:
+            json_line = json.loads(line)
+            resource_type = json_line["resourceType"]
+            resource_id   = json_line["id"]
+            post_to_fhir(line,resource_type,resource_id)
```

Comment:
Out of curiosity, what's the performance on this? I know the FHIR server has no bulk import (which absolutely sucks 🙄 ), but do we need to speed up line by line json reading? I can't imagine it's that slow so maybe a non issue.

Reply:
The parallelizing really helped, I am not sure how many lines we have to process for the ultimate upload, I also have a very old slow macbook which doesn't help. I think we won't know for sure til we get to testing on a function app